server/controllers/notification.js

```diff
@@ -64,3 +64,37 @@ exports.getNotifications = asyncHandler(async (req, res, next) => {
 
   res.status(200).json(notifications);
 });
+
+module.exports.socketCreateNotification = function (socket) {
+  socket.on('create notification', async function (data) {
+    const submissionData = data.returnData.submission;
+    const type = data.type;
+
+    const typeList = ['submission', 'message'];
+    if (!typeList.includes(type)) {
+      throw new Error('No such notification type.');
+    } else {
+      let params;
+      if (type === 'submission') {
+        const { files, _id: submissionId, contestId, userId: senderId } = submissionData;
+        // Use last file in submission array while submission featured photo is not yet set
+        const photo = files[files.length - 1];
+        const receiverId = await Contest.findById(contestId).then((contest) => {
+          return contest.userId;
+        });
+        params = { type, receiverId, senderId, contestId, submissionId, photo };
+      } else if (type === 'message') {
+        const { receiverId, userId: senderId } = submissionData;
+        params = { type, receiverId, senderId };
+      }
+
+      const notification = await Notification.create(params);
+
+      if (!notification) {
+        throw new Error('Invalid notification data');
+      }
+    }
+
+    socket.emit('notification created');
```

Comment:
You need to send the notification through an individual socket id :)


Cheatsheet: https://socket.io/docs/v3/emit-cheatsheet/index.html
Tutorial: https://www.youtube.com/watch?v=HggSXt1Hzfk&t=5312s (Watch the part where sockets are being implemented but this is for messaging)

Reply:
Thanks for the references!

Reply:
I know this is super late, but if you look at my PR I have the socketId collection available and maintained in app.js
69-74 is receiving an incoming message.  It takes the email (send up as receiver) and uses the helper getUser to get an object with the email and socketId.  
```socket.on('SEND_MESSAGE', ({ receiver, conversationId, message }) => {
    const receiverId = getUser(receiver);
    if (receiverId) {
      io.to(receiverId.socketId).emit('GET_MESSAGE', { conversationId, message });
    }
  });```

Reply:
Thanks, this helps a lot! I tried to implement something similar just now. We can check later if that works alright.